test/lit/passes/optimize-instructions-call_ref.wast

```diff
@@ -5,21 +5,27 @@
 ;; remove-unused-names is to allow fallthrough computations to work on blocks
 
 (module
- ;; CHECK:      (type $none_=>_i32 (func (result i32)))
- (type $none_=>_i32 (func (result i32)))
-
  ;; CHECK:      (type $i32_i32_=>_none (func (param i32 i32)))
  (type $i32_i32_=>_none (func (param i32 i32)))
 
+ ;; CHECK:      (type $none_=>_i32 (func (result i32)))
+ (type $none_=>_i32 (func (result i32)))
+
  ;; CHECK:      (type $none_=>_none (func))
  (type $none_=>_none (func))
 
+ ;; CHECK:      (type $i32_=>_none (func (param i32)))
+
  ;; CHECK:      (type $data_=>_none (func (param dataref)))
  (type $data_=>_none (func (param (ref data))))
 
- ;; CHECK:      (type $i32_=>_none (func (param i32)))
+ ;; CHECK:      (table $table-1 10 (ref null $i32_i32_=>_none))
+ (table $table-1 10 (ref null $i32_i32_=>_none))
+ ;; CHECK:      (elem $elem-1 (table $table-1) (i32.const 0) (ref null $i32_i32_=>_none) (ref.func $foo))
+ (elem $elem-1 (table $table-1) (i32.const 0) (ref null $i32_i32_=>_none)
+  (ref.func $foo))
 
- ;; CHECK:      (elem declare func $fallthrough-no-params $fallthrough-non-nullable $foo $return-nothing)
```

Comment:
Do you know why `$foo` was removed from the table?

Reply:
Yes, this is just the `elem declare` - it's not a real table. It just declares items that a reference is taken to them, but that are not in an actual table (this helps VMs optimize, to know that up front IIUC). As this added a use of an item, it no longer needs to be here.

(I could have worked around this to avoid a change on this line, but `elem declare` is not part of this test, it's just an artifact of the wasm text & binary formats.)